cpp/include/cudf/detail/contiguous_split.hpp

```diff
@@ -67,7 +67,7 @@ class metadata_builder {
    * @brief Destructor that will be implemented as default, required because metadata_builder_impl
    * is incomplete at this stage.
    */
-  ~metadata_builder() = default;
+  ~metadata_builder();
```

Comment:
so I am surprised this is working. I thought that because `metadata_builder` is used from both `pack.cpp` and `contiguous_split.cu` I would get compilation errors if the destructor was defined inside `pack.cpp`. I am good with the change, just curious + the compiler never lies ;)

Reply:
ah it's the other way around ;)  https://github.com/rapidsai/cudf/pull/13260/files#r1181922412